src/hooks/useWeather.jsx

```diff
@@ -0,0 +1,36 @@
+import axios from "axios";
+import {useEffect, useState} from "react";
+import {API_KEY, API_URL} from "../constants/const.ts";
+
+export function useWeather(lat, lon) {
+    const [weatherData, setWeatherData] = useState(null);
+    const [icon, setIcon] = useState(null);
+
+    useEffect(() => {
+        if (lat && lon) {
+            fetchData(lat, lon);
+        }
+    }, [lat, lon]);
+
+    async function fetchData(lat, lon) {
+        const apiUrl = `${API_URL}?lat=${lat}&lon=${lon}&appid=${API_KEY}&units=metric`;
+        const response = await axios.get(apiUrl);
+        setWeatherData(response.data);
+
+        if (response.data.weather && response.data.weather.length > 0) {
+            const iconPath = response.data.weather[0].icon;
+            const iconUrl = `https://openweathermap.org/img/wn/${iconPath}@2x.png`;
+            setIcon(iconUrl);
+        }
+    }
```

Comment:
create separate folder for fetch (get, post) requests
e.g
API => requestByCoord => index.ts
API => requestByName => index.ts

`export const geWeatherByCoords = async () => {
  return axios.get(apiUrl)
}`